src/server_state.cpp

```diff
@@ -764,9 +764,15 @@ void wsrep::server_state::on_primary_view(
     }
     else
     {
-        if (state_ == s_joiner)
+        // There are 2 possible states in which we can be here
+        // 1. s_joined when we have already done initialization
+        // 2. s_initialized when initialization hasn't been done
+        if (state_ == s_joined || state_ == s_initialized)
```

Comment:
The condition was earlier `if (state_ == s_joiner)`, wasn't this branch ever executed?

Reply:
Looking at this function i think state s_joiner will be never set.